docs/src/main/tut/typeclasses.md

```diff
@@ -0,0 +1,73 @@
+The typeclass pattern is a technique borrowed from haskell. You can
+think of it as an "interface" in the java sense, but universally
+quantified over some type. In scala they are encoded as a trait that
+has one type variable, such as:
```

Comment:
I also use multi-parameter type classes, for encoding some relationship between types.  (I think I played with 6 different kinds before settling on my 3-parameter TC.). Of course, then you can't use context bounds, unfortunately.

P.S. I also winced at the phrase "universally quantified" even though it's accurate.  How noob friendly would we like these docs to be?